ambry-router/src/test/java/com.github.ambry.router/MockServerLayout.java

```diff
@@ -45,9 +48,6 @@ public MockServerLayout(ClusterMap clusterMap) {
    */
   public synchronized MockServer getMockServer(String host, int port) {
     DataNodeId dataNodeId = clustermap.getDataNodeId(host, port);
-    if (!mockServers.containsKey(dataNodeId)) {
-      mockServers.put(dataNodeId, new MockServer());
```

Comment:
There are chances that this could throw NPE. What incase the caller passes in some random host and port (programmatic error)